### PR TITLE
Protect exts_lists from templating in dump method...

### PR DESCRIPTION
**Pull request by ocaisa**
(since they are templated internally themselves)

### CODE REVIEW

**Comment by ocaisa** [comment]
Fixes #2618

**Comment by ocaisa** [comment]
Hmm, the new test isn't broken before I add the fix, not sure what's going on there...but I confirm that it does fix the error described in #2618

**Comment by ocaisa** [comment]
@boegel I cannot get this additional test to break, I can see the difference in the outputs with `--try-toolchain` but the templating is not as aggressive for some reason in the Python case I included

**Comment by ocaisa** [comment]
Ok, if I look directly at ectxt in the test I can see the fix works, but this is not causing problems when parsing. I tried a few things to get the test to break but I am out of ideas.

**Comment by ocaisa** [comment]
Ok, the test now fails without the fix so this is good to go. I don't understand why templating needs to be switched off in the parsed easyconfig but it at least catches the error and seems to be what happens in the main trunk. I would suspect it is because that there is nested templating in `exts_list` but maybe @boegel can explain.

File: easybuild/framework/easyconfig/format/format.py [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
 # values for these keys will not be templated in dump()
 EXCLUDED_KEYS_REPLACE_TEMPLATES = ['description', 'easyblock', 'homepage', 'name', 'toolchain', 'version'] \
-                                  + DEPENDENCY_PARAMETERS
+                                  + ['exts_list'] + DEPENDENCY_PARAMETERS
```
Comment (boegel) [inline easybuild/framework/easyconfig/format/format.py]: @ocaisa Let's keep this sorted alphabetically, it'll be way easier to tell whether a particular parameter is included or not that way upon visual inspection...

Done in https://github.com/ocaisa/easybuild-framework/pull/38

**Comment by boegel** [comment]
@ocaisa To answer your question: I think we deliberately don't resolve things like `%(namelower)s` for extensions because of the ambiguity w.r.t. main `name` or extension name...

**Comment by boegel** [comment]
Going in, thanks @ocaisa!

**Pull request merged**